### PR TITLE
[18.0][MIG] product_state: Migration to 18.0

### PR DESCRIPTION
**Pull request by natuan9**

### CODE REVIEW

**Comment by LucasTran380381** [comment]
Could you fix test

I see failed due to warning log?

**Comment by LucasTran380381** [comment]
I suggest fix this warning ( just use odoo-checklog for non-test-case module)

**Comment by natuan9** [comment]
Hi @LucasTran380381 , test issue has been fixed, pls help to review it

File: product_state/views/product_template_views.xml [hunk 1]
```diff
@@ -46,7 +46,7 @@
         <field name="priority">50</field>
         <field name="arch" type="xml">
             <field name="type" position="after">
-                <field name="state" />
+                <field name="state" readonly="True" />
```
Comment (LucasTran380381) [inline product_state/views/product_template_views.xml]: why you add `readonly` field `state` is computed field => it readonly by default ?

File: product_state/views/product_template_views.xml [hunk 2]
```diff
@@ -35,7 +35,7 @@
                     widget="statusbar"
                     options="{'clickable': '1'}"
                 />
-                <field name="state" invisible="1" />
+                <field name="state" invisible="1" readonly="True" />
```
Comment (LucasTran380381) [inline product_state/views/product_template_views.xml]: It see field `state` is invisible why  could not removed it ?

**Review by LucasTran380381 — CHANGES_REQUESTED** [review]

**Review by LucasTran380381 — APPROVED** [review]

**Comment by thienvh332** [comment]
Hi @natuan9 , Your new `ir.model.access.csv` is blocking internal users from accessing the product. Please update this issue.
image

**Pull request closed without merging**